xproc/src/main/xml/specification.xml

```diff
@@ -3768,6 +3768,39 @@ linkend="initiating">static analysis</link> of the pipeline.</para>
 processing occurs after XML parsing and has no effect on well-formedness
 or validation errors which will be reported in the usual way.</para>
 </note>
+
+<para>Deadlock situations can arise if two or more 
+<tag class="attribute">[p:]use-when</tag> expressions depend on each other.
+Consider, for example:</para>
+
+<programlisting language="xml"><![CDATA[<p:declare-step type="ex:A" use-when="p:step-available('ex:B')">
+  …
+</p:declare-step>
+
+<p:declare-step type="ex:B" use-when="p:step-available('ex:A')">
+  …
+</p:declare-step>]]></programlisting>
+
+<para>It is not possible for the processor to determine if
+<code>ex:A</code> should be declared without first determining if
+<code>ex:B</code> should be declared, and vice versa.</para>
+
+<para><error code="S0115">It is a <glossterm>static error</glossterm>
+if two or more elements are contained within a deadlocked network of
+<tag class="attribute">[p:]use-when</tag> expressions.</error> In
+order to avoid deadlock, there must exist an order in which every
+expression can be resolved without reference to an expression that
+occurs after it in the ordering.</para>
+
+<para>Processors may be required to evaluate expressions in an
```

Comment:
may be required → may be allowed?

Reply:
I think required is correct. The processor has to determine an order that works, and whether or not there is one, and then evaluate the expressions in that order.